test/check-application

```diff
@@ -1447,14 +1447,73 @@ class TestFiles(testlib.MachineCase):
         b.wait_in_text(".pf-v5-c-modal-box__title-text", "bin")
         b.wait_in_text("#edit-permissions-owner", "root")
         b.wait_in_text("#edit-permissions-group", "root")
-
-        # As normal user you cannot change user/group permissions
-        b.drop_superuser()
+        b.click("button.pf-m-link")
+        b.wait_not_present(".pf-v5-c-modal-box")
 
         b.go("/files#/?path=/home/admin")
         self.assert_last_breadcrumb("admin")
         b.wait_not_present(".pf-v5-c-empty-state")
 
+        # Enclosed folder permissions
+        change_enclosed_dir = "/home/admin/Documents"
+        m.execute(f"""
+            runuser -u admin mkdir {change_enclosed_dir}
+            runuser -u admin mkdir {change_enclosed_dir}/Finance
+            runuser -u admin mkdir {change_enclosed_dir}/Work
+            runuser -u admin chmod 750 {change_enclosed_dir}/Work
+            runuser -u admin touch {change_enclosed_dir}/Finance/numbers.txt
+            runuser -u admin chmod 777 {change_enclosed_dir}/Finance/numbers.txt
+            runuser -u admin touch {change_enclosed_dir}/Finance/accounts.txt
+            runuser -u admin chmod 600 {change_enclosed_dir}/Finance/accounts.txt
+            runuser -u admin touch {change_enclosed_dir}/Work/report.txt
+            runuser -u admin chmod 750 {change_enclosed_dir}/Work/report.txt
+            runuser -u admin touch {change_enclosed_dir}/Work/notes.txt
+            runuser -u admin chmod 644 {change_enclosed_dir}/Work/notes.txt
+        """)
+
+        open_permissions_dialog("Documents")
+        b.wait_val("#edit-permissions-owner-access", "read-write")
+        b.wait_val("#edit-permissions-group-access", "read-only")
+        b.wait_val("#edit-permissions-other-access", "read-only")
+        b.click(".pf-v5-c-modal-box button.pf-m-secondary")
+        b.wait_not_present(".pf-v5-c-empty-state")
+
+        # Files with any executable bit set, get a +x in all modes when +X is passed.
+        self.assertEqual(m.execute(f"ls -ld {change_enclosed_dir}/Finance")[:10], "drwxr-xr-x")
+        self.assertEqual(m.execute(f"ls -l {change_enclosed_dir}/Finance/numbers.txt")[:10], "-rwxr-xr-x")
```

Comment:
This had me :eyes: until I realized that this was executable from the beginning, and is just badly named.